src/data/utils/dataUtils.js

```diff
@@ -24,7 +24,9 @@ export const getTpaProvider = (tpaHintProvider, primaryProviders, secondaryProvi
     secondaryProviders.forEach((provider) => {
       if (provider.id === tpaHintProvider) {
         tpaProvider = provider;
-        isSecondaryProvider = true;
+        if (provider.skipHintedLogin) {
```

Comment:
This needs in above for each as well, these providers can be primary as well.

I also suggest concatenating both arrays into one and change this function to use only one for each loop instead of two because there is no difference in the logic for both.